src/main/java/intro/bookservice/repository/book/PriceSpecificationProvider.java

```diff
@@ -0,0 +1,23 @@
+package intro.bookservice.repository.book;
+
+import intro.bookservice.model.Book;
+import intro.bookservice.repository.spec.SpecificationProvider;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PriceSpecificationProvider implements SpecificationProvider<Book> {
```

Comment:
I'm not sure searching for exact prices is very useful. Usually for prices you need to specify a range `from...to`, but let it be.